internal/services/compute/virtual_machine.go

```diff
@@ -592,27 +546,20 @@ func VirtualMachineGalleryApplicationSchema() *pluginsdk.Schema {
 	}
 }
 
-func expandVirtualMachineGalleryApplication(input []interface{}) *[]compute.VMGalleryApplication {
-	out := make([]compute.VMGalleryApplication, 0)
+func expandVirtualMachineGalleryApplication(input []interface{}) *[]virtualmachines.VMGalleryApplication {
+	out := make([]virtualmachines.VMGalleryApplication, 0)
 	if len(input) == 0 {
 		return &out
 	}
 
 	for _, v := range input {
-		packageReferenceId := v.(map[string]interface{})["version_id"].(string)
-		automaticUpgradeEnabled := v.(map[string]interface{})["automatic_upgrade_enabled"].(bool)
-		configurationReference := v.(map[string]interface{})["configuration_blob_uri"].(string)
-		order := v.(map[string]interface{})["order"].(int)
-		tag := v.(map[string]interface{})["tag"].(string)
-		treatFailureAsDeploymentFailureEnabled := v.(map[string]interface{})["treat_failure_as_deployment_failure_enabled"].(bool)
-
-		app := &compute.VMGalleryApplication{
-			PackageReferenceID:              utils.String(packageReferenceId),
-			ConfigurationReference:          utils.String(configurationReference),
-			Order:                           utils.Int32(int32(order)),
-			Tags:                            utils.String(tag),
-			EnableAutomaticUpgrade:          utils.Bool(automaticUpgradeEnabled),
-			TreatFailureAsDeploymentFailure: utils.Bool(treatFailureAsDeploymentFailureEnabled),
+		app := &virtualmachines.VMGalleryApplication{
+			PackageReferenceId:              v.(map[string]interface{})["version_id"].(string),
+			ConfigurationReference:          pointer.To(v.(map[string]interface{})["configuration_blob_uri"].(string)),
+			Order:                           pointer.To(int64(v.(map[string]interface{})["order"].(int))),
+			Tags:                            pointer.To(v.(map[string]interface{})["tag"].(string)),
+			EnableAutomaticUpgrade:          pointer.To(v.(map[string]interface{})["automatic_upgrade_enabled"].(bool)),
+			TreatFailureAsDeploymentFailure: pointer.To(v.(map[string]interface{})["treat_failure_as_deployment_failure_enabled"].(bool)),
```

Comment:
Rather than casting this multiple times, can we make this:

```suggestion
		config := v.(map[string]interface{})
		app := &virtualmachines.VMGalleryApplication{
			PackageReferenceId:              config["version_id"].(string),
			ConfigurationReference:          pointer.To(config["configuration_blob_uri"].(string)),
			Order:                           pointer.To(int64(config["order"].(int))),
			Tags:                            pointer.To(config["tag"].(string)),
			EnableAutomaticUpgrade:          pointer.To(config["automatic_upgrade_enabled"].(bool)),
			TreatFailureAsDeploymentFailure: pointer.To(config["treat_failure_as_deployment_failure_enabled"].(bool)),
```